libvast/vast/concept/convertible/data.hpp

```diff
@@ -413,14 +422,33 @@ caf::error convert(const record& src, To& dst) {
   return convert(src, dst, dst.layout);
 }
 
+// TODO: Move to a dedicated header after conversion is refactored to use
+// specialization.
+template <has_parser_v To>
+caf::error convert(std::string_view src, To& dst) {
+  const auto* f = src.begin();
+  if (!parse(f, src.end(), dst))
+    return caf::make_error(ec::convert_error,
+                           fmt::format("unable to parse \"{}\" into a {}", src,
+                                       detail::pretty_type_name(dst)));
+  return caf::none;
+}
+
 // A concept to detect whether any previously declared overloads of
 // `convert` can be used for a combination of `Type`, `From`, and `To`.
 template <class From, class To, class Type>
-concept is_concrete_convertible
+concept is_concrete_typed_convertible
   = requires(const From& src, To& dst, const Type& type) {
   { vast::convert(src, dst, type) } -> concepts::same_as<caf::error>;
```

Comment:
Don't we have some convert overloads that return bool?

```
❯ git grep "^bool convert" | wc -l
      23
```

Reply:
See my comment about specialization. For now we ignore them because they would have to be declared before the compiler parses the definition of this concept. This is normally not the case because this header is usually early in the list of includes.

Reply:
So this concept does not work for the majority of convert overloads because we should use customization points other than function overloads, e.g., niebloids or a tag_invoke-like mechanism. Gotcha. Can you add that into the comment?

Reply:
I updated the longish comment above the `IS_TYPED_CONVERTIBLE` with a suggestion for how to fix it. I'll write a story for it too.